scripts/oncallAutoAssignIssue.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+autoIssue() {
+  assigneeName=$1
+  token=$2
+  issueNum=$3
+  # Check if the user can be assigned
+  echo "Attempting to assign user $assigneeName to issue $issueNum"
+  checkAccess=$(curl --silent --output /dev/null --write-out "%{http_code}" -L -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $token" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repos/microsoft/Oryx/assignees/$assigneeName)
+  if [[ "$checkAccess" =~ ^2 ]]; then
+    res=$(curl --silent --output /dev/null --write-out "%{http_code}" -X POST "https://api.github.com/repos/microsoft/Oryx/issues/$issueNum/assignees" -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $token" -d "{\"assignees\": ["\"$assigneeName\""]}")
+    # Check for curl response status
+    if [[ "$res" =~ ^2 ]]; then
+      exit 0
+    else
+      echo "Error: Failed to assign isssue $issueNum. Check token and issue status."
+      exit 1
+    fi
+  else
+    echo "Error: Engineer Access denied."
+    return 1
+  fi
+}
+
+tryAssignIssueToCustomOncall() {
+  today=$1
+  token=$2
+  issue=$3
+  input="$CUSTOM_ONCALL_ROTATION"
+  input="${input//$'\n'/''}"
+  input="${input//$'\r'/';'}"
```

Comment:
We should make a comment here about converting the CRLF to `;` for easier string splitting later on.

I know @pauld-msft made a comment about why we're splitting on the `;` character when our documentation says to split on new lines, so documenting that we first do a conversion to separate the entries by the `;` character should clear this up.

Reply:
ah yep i missed this. 

Also, seems a little strange that we are expecting the environment variable to have newline characters in it? Is there a reason we do it this way instead of just using `;` as a separator?

Also, maybe I'm missing something, but are we sure that we can expect `\r` as a newline character for all cases? Wouldn't this make a Unix file with `LF` line endings (instead of Windows `CRLF`) not be split properly?

Reply:
unfortunately yes. so upon checking on friday, it seems like it uses CRLF instead of just `\n` or `\r`. this should be a weird behavior from how github treated new lines. but i think we could try and match for `\n\r` first and replace it to `;`. Otherwise, replace `\n` or `\r` with `;`.

Reply:
> Also, seems a little strange that we are expecting the environment variable to have newline characters in it? Is there a reason we do it this way instead of just using ; as a separator?

The main reason was for "easy-of-use"; someone looking to add a new entry to the custom on-call rotation would edit the `CUSTOM_ONCALL_ROTATION` variable in the repository's settings and just add their entry to the next line and be done. Similarly, someone looking to remove outdated entries could quickly sift through the entries since there would be one on each line. Having the semicolons between each entry wouldn't make it impossible to read, but it may make it just a tad bit more difficult to differentiate between the entries.

> Also, maybe I'm missing something, but are we sure that we can expect \r as a newline character for all cases? Wouldn't this make a Unix file with LF line endings (instead of Windows CRLF) not be split properly? I guess I'm just not sure how we are getting the newline characters into the env var, and would prefer an OS agnostic way of doing that.

We shouldn't have to worry about different line endings based on the OS since this entry is set within our repository's settings for the `CUSTOM_ONCALL_ROTATION` variable, so as long as GitHub doesn't change how they're resolving new lines in their own Variables fields, we should be OK to stick with this. If they were to change it for some reason, I'd be OK with switching to the `;` separator for our entries so that we're agnostic to any changes they make on their end.

Reply:
Thanks both for the clarifications!